client/chat/socket-handlers.ts

```diff
@@ -51,7 +51,7 @@ const eventToChatAction: EventToChatActionMap = {
 
   kick: (channelId, event) => (dispatch, getState) => {
     const { auth, chat } = getState()
-    const channelInfo = chat.idToInfo.get(channelId)
+    const channelInfo = chat.idToBasicInfo.get(channelId)
     if (!channelInfo) {
```

Comment:
This hasn't changed as of this PR but this seems extremely weird? Also shouldn't the kick and ban messages contain the information about the channel it applies to?

Reply:
Yeah, I guess it would be weird to receive a message from the server for a channel that the server thinks you're in, but the client doesn't have any information on. Still, this check seemed safer than simply forcing the Typescript to assume the info exists on the client.

I'm not sure I understand your second question; I think they do?

Reply:
> Still, this check seemed safer than simply forcing the Typescript to assume the info exists on the client.

I don't understand why you would try to be "safe" here? I wasn't suggesting sprinkling `!` on anything, that's obviously unsafe and will cause a crash in this case instead of useful behavior. The client was kicked from a channel, whether they know about it or not. You should tell them they were kicked from the channel. If you don't know what the channels name is, that's fine (I mean, it's not ideal, but it's certainly not a world-ending case), you just display a message that doesn't need the channel name. (Ideally you would display content that could load that info if it didn't have it, but that's going to be hard with a snackbar)

> I'm not sure I understand your second question; I think they do?

I mean if they contain that info then why do you need the reducer state at all? "Oh we don't have the channel name in the reducer so there's nothing I can do" is incredibly bad given the event itself has everything you need?